doc/radosgw/dynamicresharding.rst

```diff
@@ -80,6 +80,54 @@ Bucket resharding status
 
    # radosgw-admin reshard status --bucket <bucket_name>
 
+The output is a json array of 3 objects (reshard_status, new_bucket_instance_id, num_shards) per shard.
+
+For example, the output at different Dynamic Resharding stages is shown below:
+
+``1. Before resharding occurred:``
+::
+
+  [
+    {
+        "reshard_status": "none",
+        "new_bucket_instance_id": "",
+        "num_shards": -1
```

Comment:
So I just tried this on master. And instead of -1 I'm getting 18446744073709551615, which is 2^64 - 1. Also "none" is now showing "CLS_RGW_RESHARD_NONE".

It looks like some changes were made in commit 731fca4f921.

I'm thinking at the very least, rgw_admin.cc::show_reshard_status, the call to dump_unsigned should be changed to a dump_int for num_shards. And perhaps the strings in cls_rgw_types.h::to_string(const enum cls_rgw_reshard_status) should be changed as well.

That would better sync the code with your new documentation.

Reply:
As a side note, since the other commit (https://github.com/ceph/ceph/commit/731fca4f921e8227e907b204dec9f1016d66b8c3) is being (was) backported to both mimic and luminous, I'm guessing we should make sure this commit is as well. So I think that means we'll need add that to its tracker.

Reply:
Thanks, Updated the examples per the new strings at 731fca4
`cls_rgw_types.h::string to_string(const enum cls_rgw_reshard_status status)`
I will handle the switching the display of `"num_shards"` to signed in separate PR.

Reply:
PR #25538 addresses the num_shard: displaying 18446744073709551615 instead of -1

Reply:
@mkogan1 @mattbenjamin My preference would be that it display "none" rather than "CLS_RGW_RESHARD_NONE" (and the equivalent for the other states). The `radosgw-admin reshard status ...` command is for admins rather than developers. But "CLS_RGW_RESHARD_NONE" contains C/C++ artifacts, namely because it's part of an `enum` (rather than an `enum class`) we add prefixes like "CLS_RGW_RESHARD" in order to avoid conflicts in the global namespace.

So my inclination is to update the code to use "none", etc. and keep the documentation fix as it was in the first iteration. But I defer to both of you.

Reply:
Thanks, changed as suggested from displaying of `CLS_RGW*` C/C++ artifacts to displaying admin friendly artifacts